client/my-sites/plans-features-main/index.tsx

```diff
@@ -287,7 +288,11 @@ const PlansFeaturesMain = ( {
 		// `cartItemForPlan` is empty if Free plan is selected. Show `FreePlanPaidDomainDialog`
 		// in that case and exit. `FreePlanPaidDomainDialog` takes over from there.
 		// - only applicable to main onboarding flow (default `/start`)
-		if ( 'onboarding' === flowName && domainName && ! cartItemForPlan ) {
+		if (
+			( 'onboarding' === flowName || isOnboardingPMFlow( flowName ) ) &&
+			domainName &&
+			! cartItemForPlan
+		) {
```

Comment:
Non-blocking: Now that we have multiple flows needing this modal, I was wondering if we could introduce something like a `isFreePlanPaidDomainDialogEnabled` prop to be passed to `PlansFeaturesMain` from higher in the tree? So this condition would then become
```js
if ( isFreePlanPaidDomainDialogEnabled && domainName && ! cartItemForPlan )
```

Reply:
Good point. Given the discussion in p2-pdgrnI-2nV#comment-4110 and how @isatuncman-auto  described it in https://github.com/Automattic/growth-foundations/issues/63#issuecomment-1622683650. It's likely that we can't just naively make it the default behavior, so something like this would be pretty nice. 

I'll do that in a follow-up PR :)